src/m365/spfx/commands/package/package-teams-enable.ts

```diff
@@ -0,0 +1,206 @@
+          return `The supportedHost contains an invalid value. Possible supportedHost values are: ${SpfxPackageTeamsEnableCommand.allowedSupportedHosts.join(',')}`;
+        }
+
+        return true;
+      }
+    );
+  }
+
+  public async commandAction(logger: Logger, args: CommandArgs): Promise<void> {
+    let tmpDir: string | undefined = undefined;
+
+    try {
+      if (this.verbose) {
+        logger.logToStderr('Creating temp folder to save the sppkg content in');
+      }
+
+      tmpDir = fs.mkdtempSync(path.join(os.tmpdir(), 'cli-spfx'));
+
+      if (this.verbose) {
+        logger.logToStderr(`Temp folder created at path: ${tmpDir}`);
+      }
+
+      const fullPath: string = path.resolve(args.options.filePath);
+      // we need this to be able to inject mock AdmZip for testing
+      /* c8 ignore next 3 */
+      if (!this.solutionZip) {
+        this.solutionZip = new AdmZip(fullPath);
+      }
+
+      this.solutionZip.extractAllTo(tmpDir);
+      const files = fs.readdirSync(tmpDir);
+
+      let fixesApplied = false;
+
+      files.forEach(file => {
+
+        if (validation.isValidGuid(file)) {
+
+          if (this.verbose) {
+            logger.logToStderr(`Found webpart folder with id ${file}`);
+          }
+
+          const dirLocation = `${tmpDir}\\${file}`;
+          const webpartFiles = fs.readdirSync(dirLocation);
+
+          webpartFiles.forEach((wpFile: string) => {
+            if (this.verbose) {
+              logger.logToStderr(`Reading contents for file ${wpFile}`);
+            }
+
+            const fileLocation = `${dirLocation}\\${wpFile}`;
+            let fileContent = fs.readFileSync(fileLocation, { encoding: 'utf-8' });
+            const regex = new RegExp('ComponentManifest=\\"[^\\"]+\\"');
+
+            if (regex.test(fileContent)) {
+
+              const matches = fileContent.match(regex);
+              const componentManifest = matches![0];
+              const componentManifestReplaced = componentManifest.replace(/&quot;/gi, '"').replace('ComponentManifest=\"', '').slice(0, -1);
+              const parsedComponentManifest: { id: string, alias: string, supportedHosts: string[] } = JSON.parse(componentManifestReplaced);
+              const missingSupportedHosts: any[] = [];
+
+              if (args.options.supportedHost) {
+                args.options.supportedHost.split(',').forEach((supportedHost: string) => {
+                  if (!parsedComponentManifest.supportedHosts.some(existing => existing === supportedHost)) {
+                    missingSupportedHosts.push(supportedHost);
+                  }
+                });
+              }
+              else {
+                if (!parsedComponentManifest.supportedHosts.some(existing => existing === 'TeamsPersonalApp')) {
+                  missingSupportedHosts.push('TeamsPersonalApp');
+                }
+              }
```

Comment:
I was wondering. What if we could do something like this

```suggestion
              const supportedHosts = args.options.supportedHost ? args.options.supportedHost.split(',') : ['TeamsPersonalApp'];
              
              supportedHosts.forEach((supportedHost: string) => {
                if (!parsedComponentManifest.supportedHosts.some(existing => existing === supportedHost)) {
                  missingSupportedHosts.push(supportedHost);
                }
              });
```

we could reduce a bit the amount of similar logic and one if block. 
What do you think?